src/commoncode/fetch.py

```diff
@@ -76,11 +76,17 @@ def ping_url(url):
     Returns True is `url` is reachable.
     """
     import urllib2
-
-    # FIXME: if there is no 200 HTTP status, then the ULR may not be reachable.
+    # If there is no 200 HTTP status, then the URL may not be reachable.
+    http_url_error_list = [400,401,403,404,408,500,501,502,503,504,505]
+    request = urllib2.Request(url)
     try:
-        urllib2.urlopen(url)
-    except Exception:
+        response = urllib2.urlopen(request)
+    except urllib2.HTTPError as e:
+        if e.code == any(http_url_error_list):
```

Comment:
@pombredanne, what I did in the changes is for this to only have a single return code and print the HTTP response code as it returns a value of False.